### PR TITLE
[510] Removes btime dependency

### PR DESCRIPTION
**Pull request by Tastyfish**
- Instead of `TimeOfHour`, there is now `TimeOfGame` which doesn't go negative weirdly or anything.
- There is also the macro `TimeOfTick`, which is `tick_usage` scaled to real time.
- Between those, `sleepCheck()` exclusively uses the `TimeOfTick` macro, which is very cheap.
- `sleepCheck()` now also is called _every_ `SCHECK` because why not. We could adjust this if we decide that's still a terrible idea.
- The scheduler now messages admins and properly logs when processes hang, instead of outputting to world.

- This is my 200'th PR apparently. :tada:

#### Some logs where I spawned 2000 foxes from the round start and ran the game for 2 hours. IE basically the worst case scenario. (It never lagged)
Imgur
Imgur
#### 4000 foxes on the server because why not.
Imgur

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
@volundr-

For what it's worth.

File: code/game/objects/explosion.dm [hunk 1]
```diff
@@ -31,7 +31,7 @@
 		flame_range = min (MAX_EX_FLAME_RANGE, flame_range)
 
 	spawn(0)
-		var/start = TimeOfHour
+		var/watch = start_watch()
```
Comment (Fox-McCloud) [inline code/game/objects/explosion.dm]: you could just make this world.timeofday

File: code/game/objects/explosion.dm [hunk 2]
```diff
@@ -157,8 +157,7 @@
 								var/turf/throw_at = get_ranged_target_turf(I, throw_dir, throw_range)
 								I.throw_at(throw_at, throw_range, 2, no_spin = 1) //Throw it at 2 speed, this is purely visual anyway; don't spin the thrown items, it's very costly.
 */
-		var/timeofhour = TimeOfHour
-		var/took = round((start <= timeofhour) ? ((timeofhour-start)/10) : ((timeofhour-(start-36000))/10), 0.01)
+		var/took = stop_watch(watch)
```
Comment (Fox-McCloud) [inline code/game/objects/explosion.dm]: and this `(world.timeofday-start)/10`

Though I don't know which will be more accurate.

**Comment by Keilden** [comment]
Should have spawned 4000 foxes and 1 space bear.

File: code/game/objects/explosion.dm (hunk 1 above)
Comment (Tastyfish) [inline code/game/objects/explosion.dm]: timeofday...is really gross because it wraps around at midnight and becomes less precise later in the day.

**Comment by Spacemanspark** [comment]
That's horrifying.

**Pull request merged**